### PR TITLE
Remove DiagnosticAnalyzerTestBase

### LINKED ISSUE #3089: Remove class DiagnosticAnalyzerTestBase
As a follow-up action of #3052 and #3057 we could get rid of the `DiagnosticAnalyzerTestBase` class.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3089

### CODE REVIEW

File: src/Utilities.UnitTests/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisTests.cs [hunk 1]
```diff
@@ -1270,5 +1272,160 @@ public PropertySetAnalysisTests(ITestOutputHelper output)
         {
             this.TestOutput = output;
         }
+
```
Comment (Evangelink) [inline src/Utilities.UnitTests/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisTests.cs]: All the following lines are copy-pasted from `DiagnosticAnalyzerTestBase` as I couldn't find a way to achieve the specific checks that are done in this class.

@sharwell If you have any good idea, please feel free to share and I'll take care to do the update.

Comment (mavasani) [reply src/Utilities.UnitTests/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisTests.cs]: @Evangelink Can you please retarget the changes to this file 2.9.x branch? `PropertySetAnalysisTests` are used by security analyzers, and will likely be updated in 2.9.x branch more often then in master.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotAddArchiveItemPathToTheTargetFileSystemPathTests.cs [hunk 2]
```diff
@@ -228,7 +227,7 @@ public void TestMethod(ZipArchiveEntry zipArchiveEntry)
     {
         zipArchiveEntry.ExtractToFile(zipArchiveEntry.FullName);
     }
-}", DiagnosticAnalyzerTestBase.GetEditorConfigAdditionalFile(editorConfigText), expected);
```
Comment (mavasani) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotAddArchiveItemPathToTheTargetFileSystemPathTests.cs]: Kindly move all the changes to security analyzers to 2.9.x branch.

**Comment by Evangelink** [comment]
@mavasani Removing the base class in the 2.9.x branch leads to LOT of changes as all of the other base class removal I did on master are not done on the 2.9.x branch. What do you want me to do? Duplicate all the base class removal? Only remove the base class declaration on files I have changed in this PR?

**Comment by mavasani** [comment]
@Evangelink I meant the changes to the security analysis test files should go in 2.9.x branch. Once that flows back into master, the removal of DiagnosticAnalyzerTestBase can happen in master itself with this PR.

**Comment by mavasani** [comment]
@Evangelink want to reopen this PR with merged changes from 2.9.x branch?

**Comment by mavasani** [comment]
@Evangelink Can you please merge master again and resolve conflicts?

**Comment by Evangelink** [comment]
Every time I'm double checking the merge I did everything's alright but when I don't it fails... Yet I am always optimistic xD

**Pull request merged**